sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -285,10 +289,29 @@ def get_all_options(
       cls._add_argparse_args(parser)  # pylint: disable=protected-access
     if add_extra_args_fn:
       add_extra_args_fn(parser)
+
     known_args, unknown_args = parser.parse_known_args(self._flags)
-    if unknown_args:
-      _LOGGER.warning("Discarding unparseable args: %s", unknown_args)
-    result = vars(known_args)
+    if retain_unknown_options:
+      i = 0
+      while i < len(unknown_args):
+        # Treat all unary flags as booleans, and all binary argument values as
+        # strings.
+        if i + 1 >= len(unknown_args) or unknown_args[i + 1].startswith('-'):
+          split = unknown_args[i].rsplit('=')
```

Comment:
Why `rsplit`? Shouldn't this be: 
```suggestion
          split = unknown_args[i].split('=', 1)
```

?

Reply:
Otherwise this will break options like `experiments=state_cache_size=1`.

Reply:
The split is only used to get the argument name, and the rest of the split isn't used.

Reply:
Comment still applies. Splitting should be performed from the left, not the right. At most one split has to be performed.

Reply:
Oh, I see what you mean now. I will commit your suggestion to avoid confusion. However, I don't think we should support unrecognized `append` arguments. I don't want to have to infer complex argument types. As for the specific case of `experiments`, that will not be a problem because `experiments` is already recognized by the parser.

Reply:
I will merge this now. If we need to support `append` arguments for some reason, we should be able to add that later.

Reply:
I was only referring to options values containing `=` which can be the case independently of append options. That is fixed now. So all good.